lib/code_corps_web/controllers/github_event_controller.ex

```diff
@@ -43,14 +45,29 @@ defmodule CodeCorpsWeb.GithubEventController do
 
   @spec create(Conn.t, map) :: Conn.t
   def create(%Conn{} = conn, %{} = payload) do
-    type = conn |> get_event_type
-    delivery_id = conn |> get_delivery_id()
+    type = conn |> event_type()
+    delivery_id = conn |> delivery_id()
     action = payload |> Map.get("action", "")
-    event_support = type |> EventSupport.status(action)
-    event_support |> process_event(type, delivery_id, payload)
+    event_support =
+      if should_process?(payload) do
+        process_status = type |> EventSupport.status(action)
+        process_status |> process_event(type, delivery_id, payload)
+        process_status
+      else
+        :ignored
+      end
     conn |> respond_to_webhook(event_support)
   end
 
+  @spec should_process?(map) :: boolean
+  def should_process?(%{"repository" => %{"id" => repository_id}}) do
+    query = from repo in GithubRepo,
+      where: repo.github_id == ^repository_id,
+      where: not(is_nil(repo.project_id))
+    Repo.one(query) != nil
+  end
+  def should_process?(_), do: true
```

Comment:
We probably want to make this fn private.

Reply:
We absolutely do, good catch.